control/tap-ctl-unpause.c

```diff
@@ -54,8 +55,8 @@ tap_ctl_unpause(const int id, const int minor, const char *params, int flags,
 	message.u.params.flags = flags;
 
 	if (params)
-		strncpy(message.u.params.path, params,
-				sizeof(message.u.params.path) - 1);
+		safe_strncpy(message.u.params.path, params,
+			     sizeof(message.u.params.path));
 	if (secondary) {
 		err = snprintf(message.u.params.secondary,
 				sizeof(message.u.params.secondary) - 1, "%s",
```

Comment:
Not related to this PR, but there's an off-by-one here. If the str to write is `sizeof(message.u.params.secondary) - 1` chars long (excluding terminator), then snprintf will truncate it to `sizeof(message.u.params.secondary) - 2` chars long (+ a nul terminator) and return `sizeof(message.u.params.secondary) - 1`. The return value is not caught by the subsequent overflow check.

Reply:
Good catch, might as well fix that as well while we're there.